pkg/curatedpackages/curatedpackages.go

```diff
@@ -0,0 +1,61 @@
+package curatedpackages
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/helm"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+func NewRegistry(ctx context.Context, deps *dependencies.Dependencies, registry, kubeVersion string) (BundleRegistry, error) {
+	if registry != "" {
+		username, password, err := helm.ReadRegistryCredentials()
```

Comment:
Imo we should either inject the configuration (user/password) or inject the configurator reader in the constructor.